contracts/mock/Foreign20.sol

```diff
@@ -14,14 +14,12 @@ import { MockNativeMetaTransaction } from "./MockNativeMetaTransaction.sol";
  *
  * @notice Mock ERC-(20) NFT for Unit Testing
  */
-contract Foreign20 is ERC20Pausable, MockNativeMetaTransaction {
+contract Foreign20 is ERC20Pausable {
```

Comment:
@levalleux-ludo but there is no need to inherit this contract to support MetaTx on this Foreign20 contract. The only needed piece of code is override of `_msgSender`, which continues on the code. The `MockNativeMetaTransaction` is for the forward contract, not the receptor.

Reply:
@anajuliabit the Foreign20 has been upgraded to support meta-transactions the same way the other child token contracts on polygon support meta-transactions.
In theory it's possible to use a forwarder to relay meta-transaction to a contract, but only if the contract has been implemented to do so, hasn't it?
And that's not how the child tokens have been implemented on Polygon. 
So I think we must keep the same logic as before

Reply:
> but only if the contract has been implemented to do so, hasn't it?

Which contract do you mean? As far as I know, the receptor contract only needs to override `_msgSender`, and the forwarder contract is the one that needs to have `executeMetaTransaction` function. 
Having the receptor and executor logic on the same contract looks dangerous to me because the ERC20 contract would need to keep native token on its balance to exec meta transactions. 
This `UChieldERC20` is the pattern adopted by solidity community? 
Jut asking because OZ keeps forwarder and receptor separated https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/metatx

Reply:
"would need to keep native token on its balance to exec meta transactions": sorry I don't understand

Reply:
Just to give context: our need is to relay to the token contract a meta-transaction, for instance an Approve method, signed by the user to allow the protocol to transfer token on their behalf.
To do it, all the token contracts we're working with have a method "executeMetaTransaction" that we're calling through Biconomy.
We don't use any Forwarder

Reply:
Just to mention, the meta transaction at the protocol level work the same way (implements its own method executeMetaTransaction), so no need to use a Forwarder contract

Reply:
Btw I was thinking we are not going to use any forwarder for our meta-transactions, because it requires to trust the forwarder contract

Reply:
You are right. I got confused because both Biconomy docs and OZ use separate contracts
<img width="829" alt="Screen Shot 2023-01-20 at 11 49 32" src="https://user-images.githubusercontent.com/50335980/213727430-ae29deae-6008-4f2f-8712-4a6bb6b28f4a.png">

I'll revert the changes on MockNativeMetaTransaction. Do you think it's better to use the same pattern used for ChildERC20 on Boson Voucher as well (inherited native metatx directly on BosonVoucher)? Otherwise, we would need to have a trusted forwarder. cc @zajck @mischat

Reply:
No worries
IMHO we should use the same pattern in BosonVoucher than in ChildERC20

Reply:
We have decided to use OZ approach for now, as it is simpler and wouldn't require a full audit since we are inheriting from OZ. We can implement the Child token approach later if we want to do it